src/deploy-cromwell-on-azure/Deployer.cs

```diff
@@ -96,7 +99,9 @@ public async Task<int> DeployAsync()
 
                 tokenCredentials = new TokenCredentials(new RefreshableAzureServiceTokenProvider("https://management.azure.com/"));
                 azureCredentials = new AzureCredentials(tokenCredentials, null, null, AzureEnvironment.AzureGlobalCloud);
-                azureClient = GetAzureClient(azureCredentials);
+                azureSubscriptionClient = GetAzureSubscriptionClient(azureCredentials);
+                azureGeneralClient = GetAzureGeneralClient(azureCredentials);
```

Comment:
azureClient is descriptive enough without the word General.
GetAzureSubscriptionClient() calls GetAzureGeneralClient() and sets the property azureGeneralClient as the side effect. Get methods should not have side effects!
Then line 103 calls GetAzureGeneralClient() and sets the azureGeneralClient again.
Instead drop the GetAzureSubscriptionClient() method and modify lines 102 and 103 as follows:
azureClient = GetAzureClient(azureCredentials);
azureSubscriptionClient = azureClient.WithSubscription(configuration.SubscriptionId);

Reply:
I have renamed azureGeneralClient to azureClient.  Then deleted GetAzureSubscriptionClient and the subscriptionClient is set on line 103.